api/models/User.js

```diff
@@ -161,6 +161,72 @@ module.exports = bookshelf.Model.extend(merge({
     .digest('hex')
   },
 
+  reactivate: function () {
+    return this.save({ active: true })
+  },
+
+  deactivate: async function (sessionId) {
+    Queue.classMethod('User', 'clearSessionsFor', { userId: this.get('user_id'), sessionId })
+    return this.save({ active: false })
+  },
+
+  sanelyDeleteUser: async function ({ sessionId, transacting = {} }) {
+    const deletedUserResult = await bookshelf.knex.raw('select id from users where email = "deleted@hylo.com"')
+    const deletedUserId = deletedUserResult.rows[0].id
+    Queue.classMethod('User', 'clearSessionsFor', { userId: this.get('user_id'), sessionId })
+    /* 
+      ### List of things to be done on account deletion ###
+
+      - zero out content of their posts and comments
+      - change the ownership of their posts and comments to the deletedUser record
+      - remove other references to their user_id
+      - delete their user record!
+    */
+
+    const query = `
+    BEGIN;
+    UPDATE posts SET name = 'Post by deleted user', description = '', active = false, location = NULL, location_id = NULL WHERE user_id = ${this.id};
+    UPDATE posts_users SET user_id = ${deletedUserId} WHERE user_id = ${this.id};
+    DELETE FROM user_connections WHERE (user_id = ${this.id}) OR (other_user_id = ${this.id});
+
+    UPDATE groups SET created_by_id = ${deletedUserId} WHERE created_by_id = ${this.id};
+    UPDATE comments SET deactivated_by_id = ${deletedUserId} WHERE deactivated_by_id = ${this.id};
+    UPDATE comments SET text = 'Comment by deleted user' WHERE user_id = ${this.id};
+
+    UPDATE follows set added_by_id = ${deletedUserId} WHERE added_by_id = ${this.id};
+    UPDATE groups_tags set user_id = ${deletedUserId} WHERE user_id = ${this.id};
+    DELETE FROM thanks WHERE comment_id in (select id from comments WHERE user_id = ${this.id});
+    DELETE FROM thanks WHERE thanked_by_id = ${this.id};
+    DELETE FROM notifications WHERE activity_id in (select id from activities WHERE reader_id = ${this.id});
+    DELETE FROM notifications WHERE activity_id in (select id from activities WHERE actor_id = ${this.id});
+    DELETE FROM activities WHERE actor_id = ${this.id};
+    DELETE FROM activities WHERE reader_id = ${this.id};
+
+    DELETE FROM contributions WHERE user_id = ${this.id};
+    DELETE FROM devices WHERE user_id = ${this.id};
+    DELETE FROM group_invites WHERE used_by_id = ${this.id};
+    DELETE FROM group_invites WHERE invited_by_id = ${this.id};
```

Comment:
this one im not sure about, should the invite be rescinded?

Reply:
I think this is what I copied from the delete user DB procedure, so I don't have a strong opinion. If we don't want to rescind, I suppose we'll have to switch the `id` over to the deleted user.

Reply:
hmm, maybe we should ask what people think, im not sure what is right here